### ISSUE TITLE
Blank response from kamal-proxy, bug in logging middleware?

### ISSUE THREAD

**Issue by aarroisi** [basecamp/kamal-proxy#49]
I got this error using kamal-proxy version 0.8 with this error message showed up:
```
{"time":"2024-10-09T15:55:46.963670551Z","level":"INFO","msg":"http: superfluous response.WriteHeader call from github.com/basecamp/kamal-proxy/internal/server.(*loggerResponseWriter).WriteHeader (logging_middleware.go:132)"}
```

As a result, the response is empty and app is unusable.

The curious thing is, I have 4 apps running behind the same proxy, but the issue only exist in one of the apps.

**Comment by aarroisi** [comment]
Some information:
- When I go back to 0.7, it works again.
- The app is a Node.js app (Nuxt 2).

Other peculiar things I found:
- If opened with curl, it returns something (the html of the page, but I don't know if it contains any error). The "superfluous response.WriteHeader" log still showed up in kamal-proxy.
- If opened with browser, nothing is returned in the network or browser console. 0 data. The "superfluous response.WriteHeader" log showed up in kamal-proxy.

When I go back to 0.7.0, all of those are gone and app is working perfectly.

**Comment by kevinmcconnell** [comment]
@aarroisi thanks for reporting. This is actually the same bug as #47, we're shipping a fix for it right now.

**Comment by kevinmcconnell** [comment]
@aarroisi we've just released a new Kamal version (2.2.1) which should fix this.

I'm closing this issue as a dup of #47 but if you still see any issues after upgrading to 2.2.1 please let me know.
